Automattic/wp-calypso
# Domains: Use new domain design in site domains view

#### Changes proposed in this Pull Request

This updates the site domains view to use the new design when the `manage/all-domains` flag is enabled – along with some fixes – though the work is far from complete. For the time being changing the primary domain is not possible, but since it's behind a flag it's not an issue.

<img width="1062" alt="Screenshot 2020-07-08 at 6 18 08 PM" src="https://user-images.githubusercontent.com/13062352/86944137-9b443a00-c147-11ea-90b7-6a87ebad1981.png">

<img width="281" alt="Screenshot 2020-07-08 at 6 18 18 PM" src="https://user-images.githubusercontent.com/13062352/86944152-9ed7c100-c147-11ea-8d33-459726ac5758.png">


#### Testing instructions

- The /domains/manage view of all domains should not have changed, apart from fixes such as showing correct values for email or other columns
- Going to a site -> Manage -> Domains should show the new design as in the screenshot
- Verify the logic for transfer lock/privacy/auto-renew/email makes sense
